src/wolfio.c

```diff
@@ -2714,4 +2714,382 @@ int wolfSSL_SetIO_LwIP(WOLFSSL* ssl, void* pcb,
+        XMEMCPY(ctx->buf_ptr, &ctx->frame.data[1], frame_len);
+        ctx->buf_ptr += frame_len;
+        ctx->buf_length += frame_len;
+        data_size -= frame_len;
+
+        /* Sequence is 0 - 15 (first 0 is implied for first packet */
+        ctx->sequence++;
+        if (ctx->sequence > ISOTP_MAX_SEQUENCE_COUNTER) {
+            ctx->sequence = 0;
+        }
+    }
+    return ctx->buf_length;
+
+}
+
+/* The wolfSSL receive callback, needs to buffer because we need to grab all
+ * incoming data, even if wolfSSL doesn't want it all yet */
+int ISOTP_Receive(WOLFSSL* ssl, char* buf, int sz, void* ctx)
+{
+    (void) ssl;
+    enum isotp_frame_type type;
+    int ret;
+    struct isotp_wolfssl_ctx *isotp_ctx = (struct isotp_wolfssl_ctx*)ctx;
+
+    /* Is buffer empty? If so, fill it */
+    if (!isotp_ctx->receive_buffer_len) {
+            /* Can't send whilst we are receiving */
+        if (isotp_ctx->state != ISOTP_CONN_STATE_IDLE) {
+            return WOLFSSL_ERROR_WANT_READ;
+        }
+        isotp_ctx->state = ISOTP_CONN_STATE_RECEIVING;
+        do {
+            ret = isotp_ctx->recv_fn(&isotp_ctx->frame, isotp_ctx->arg,
+                    ISOTP_DEFAULT_TIMEOUT);
+        } while (ret == 0);
+        if (ret == 0) {
+            isotp_ctx->state = ISOTP_CONN_STATE_IDLE;
+            return WOLFSSL_CBIO_ERR_TIMEOUT;
+        } else if (ret < 0) {
+            isotp_ctx->state = ISOTP_CONN_STATE_IDLE;
+            WOLFSSL_MSG("ISO-TP receive error");
+            return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+
+        type = isotp_ctx->frame.data[0] >> 4;
+
+        if (type == ISOTP_FRAME_TYPE_SINGLE) {
+            isotp_ctx->receive_buffer_len =
+                isotp_receive_single_frame(isotp_ctx);
+        } else if (type == ISOTP_FRAME_TYPE_FIRST) {
+            isotp_ctx->receive_buffer_len =
+                isotp_receive_multi_frame(isotp_ctx);
+        } else {
+            /* Should never get here */
+            isotp_ctx->state = ISOTP_CONN_STATE_IDLE;
+            WOLFSSL_MSG("ISO-TP frames out of sequence");
+            return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+        if (isotp_ctx->receive_buffer_len <= 1) {
+            isotp_ctx->state = ISOTP_CONN_STATE_IDLE;
+            return isotp_ctx->receive_buffer_len;
+        } else {
+            isotp_ctx->receive_buffer_ptr = isotp_ctx->receive_buffer;
+        }
+        isotp_ctx->state = ISOTP_CONN_STATE_IDLE;
+    }
+
+    /* Return from the buffer */
+    if (isotp_ctx->receive_buffer_len >= sz) {
+        XMEMCPY(buf, isotp_ctx->receive_buffer_ptr, sz);
+        isotp_ctx->receive_buffer_ptr+= sz;
+        isotp_ctx->receive_buffer_len-= sz;
+        return sz;
+    } else {
+        XMEMCPY(buf, isotp_ctx->receive_buffer_ptr,
+                isotp_ctx->receive_buffer_len);
+        sz = isotp_ctx->receive_buffer_len;
+        isotp_ctx->receive_buffer_len = 0;
+        return sz;
+    }
+}
+
+/* Setup ISPTP
```

Comment:
Can you move or copy this Doxy to the documentation dir so it goes into the manual?